src/core/rewrite.ml

```diff
@@ -679,29 +698,17 @@ let symmetry : Sig_state.t -> popt -> Proof.t -> term = fun ss pos ps ->
   let (g_env, g_type) = Proof.focus_goal pos ps in
   (* Check that the type of [g] is of the form “P (eq a l r)”. *)
   let (a, l, r) = get_eq_data pos cfg g_type in
-  (* NOTE The proofterm is “eqind a r l M (λx,eq a l x) (refl a l)”. *)
-  (* We create a new metavariable (“M” in the above). *)
+  (* We create a new metavariable of type [P (eq a r l)]. *)
   let meta_type =
     Appl(Symb cfg.symb_P, (add_args (Symb cfg.symb_eq) [a; r; l])) in
   let meta_term = make_meta (Env.to_ctxt g_env) meta_type in
-  (* We build the predicate (“λx, eq a r x” in the above). *)
-  let pred =
-    let x = Bindlib.new_var mkfree "X" in
-    let pred = add_args (Symb cfg.symb_eq) [a; l; Vari(x)] in
-    let pred = Bindlib.unbox (Bindlib.bind_var x (lift pred)) in
-    Abst(Appl(Symb cfg.symb_T, a), pred)
-  in
-  (* We build the proof term. *)
-  let refl_a_l = add_args (Symb cfg.symb_refl) [a; l] in
-  let term =
-    add_args (Symb cfg.symb_eqind) [a; r; l; meta_term; pred; refl_a_l] in
+  let term = swap cfg a r l meta_term in
```

Comment:
It looks a bit unclear to me, why is there a call to `swap` here?

Reply:
Well, this is the goal of the symmetry tactic to turn a goal of the form x=y into a goal of the form y=x.